src/crc32.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2024 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <csp/csp.h>
+#include <csp/csp_crc32.h>
+#include <csp/csp_id.h>
+#include <endian.h>
+
+struct crc_fixture {
+	csp_packet_t *buf;
+};
+
+static struct crc_fixture crc_fixture;
+
+static void *setup(void)
+{
+	csp_init();
+
+	return &crc_fixture;
+}
+
+static void before(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	fixture->buf = csp_buffer_get(0);
+	zassert_true(fixture->buf != NULL, NULL);
+}
+
+static void after(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	csp_buffer_free(fixture->buf);
+}
+
+ZTEST_F(crc, test_crc_verify_without_crc)
+{
+	csp_packet_t *packet = fixture->buf;
+	int ret;
+
+	/* Check size error */
+	packet->length = 0;
+	ret = csp_crc32_verify(packet);
+	zassert_equal(ret, CSP_ERR_CRC32, "Error on csp_crc32_verify");
```

Comment:
Please add the causes to the content of the error messages so that it's clear which error is occured.